datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -491,6 +491,55 @@ impl GroupedHashAggregateStream {
         Ok(())
     }
 
+    // Update the accumulator results, according to row_aggr_state.
+    fn update_accumulators_using_scalar(
+        &mut self,
+        groups_with_rows: &[usize],
+        row_values: &[Vec<ArrayRef>],
+        row_filter_values: &[Option<ArrayRef>],
+    ) -> Result<()> {
+        let filter_bool_array = row_filter_values
+            .iter()
+            .map(|filter_opt| match filter_opt {
+                Some(f) => Ok(Some(as_boolean_array(f)?)),
```

Comment:
Seems you don't need to wrap the result in `Ok` / `Result` here.

Reply:
the `as_boolean_array` returns a Result here.